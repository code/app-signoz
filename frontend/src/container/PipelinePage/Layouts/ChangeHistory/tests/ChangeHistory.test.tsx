import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';
import { Pipeline } from 'types/api/pipeline/def';
import { DataTypes } from 'types/api/queryBuilder/queryAutocompleteResponse';

import ChangeHistory from '../index';

const queryClient = new QueryClient({
	defaultOptions: {
		queries: {
			refetchOnWindowFocus: false,
		},
	},
});

describe('ChangeHistory test', () => {
	it('should render changeHistory correctly', () => {
		// const pipelinedata: Pipeline = {
		// 	active: true,
		// 	createdBy: 'admin',
		// 	deployResult: 'random_data',
		// 	deployStatus: 'random_data',
		// 	disabled: false,
		// 	elementType: 'random_data',
		// 	history: [],
		// 	id: v4(),
		// 	isValid: true,
		// 	lastConf: 'random_data',
		// 	lastHash: 'random_data',
		// 	pipelines: [],
		// 	version: 1,
		// };
		const pipelinedata: Pipeline = {
			id: '8e196a3c-8737-46d2-8b62-264353b7b9aa',
			version: 24,
			elementType: 'log_pipelines',
			active: false,
			is_valid: false,
			disabled: false,
			deployStatus: 'DEPLOYED',
			deployResult: 'Deployment was successful',
			lastHash: 'log_pipelines:24',
			lastConf:
				'[{"id":"1917badc-580c-42fe-a959-cdf895d4f27b","orderId":1,"name":"hotrod logs parser","alias":"hotrodlogsparser","description":"Trying to test Logs Pipeline feature","enabled":true,"filter":{"op":"AND","items":[{"key":{"key":"container_name","dataType":"string","type":"tag","isColumn":false,"isJSON":false},"value":"hotrod","op":"="}]},"config":[{"type":"regex_parser","id":"parsetext(regex)","output":"parseattribsjson","on_error":"send","orderId":1,"enabled":true,"name":"parse text (regex)","parse_to":"attributes","regex":".+\\\\t+(?P\\u003clog_level\\u003e.+)\\\\t+(?P\\u003clocation\\u003e.+)\\\\t+(?P\\u003cmessage\\u003e.+)\\\\t+(?P\\u003cattribs_json\\u003e.+)","parse_from":"body"},{"type":"json_parser","id":"parseattribsjson","output":"removetempattribs_json","orderId":2,"enabled":true,"name":"parse attribs json","parse_to":"attributes","parse_from":"attributes.attribs_json"},{"type":"remove","id":"removetempattribs_json","output":"c2062723-895e-4614-ba38-29c5d5ee5927","orderId":3,"enabled":true,"name":"remove temp attribs_json","field":"attributes.attribs_json"},{"type":"add","id":"c2062723-895e-4614-ba38-29c5d5ee5927","orderId":4,"enabled":true,"name":"test add ","field":"resource[\\"container.name\\"]","value":"hotrod"}],"createdBy":"prashant@signoz.io","createdAt":"2024-01-02T13:56:02.858300964Z"},{"id":"6aea8303-4a62-4c62-ba07-f16ff8f2174a","orderId":2,"name":"Logs Parser - Pranay - Customer Service","alias":"LogsParser-Pranay-CustomerService","description":"Trying to test Logs Pipeline feature","enabled":true,"filter":{"op":"AND","items":[{"key":{"key":"service","dataType":"string","type":"tag","isColumn":false,"isJSON":false},"value":"customer","op":"="}]},"config":[{"type":"grok_parser","id":"TestPranay","on_error":"send","orderId":1,"enabled":true,"name":"Test Pranay","parse_to":"attributes","pattern":"^%{DATE:date}Z INFO customer/database.go:73 Loading customer {\\"service\\": \\"customer\\", \\"component\\": \\"mysql\\", \\"trace_id\\": \\"227e184cb1263724\\", \\"span_id\\": \\"1427a3fcad8b1514\\", \\"customer_id\\": \\"567\\"}","parse_from":"body"}],"createdBy":"prashant@signoz.io","createdAt":"2024-01-02T13:56:02.863764227Z"}]',
			createdBy: '5e9681b1-16c9-4f7a-9a62-48904d73d9c9',
			pipelines: [
				{
					id: '1917badc-580c-42fe-a959-cdf895d4f27b',
					orderId: 1,
					name: 'hotrod logs parser',
					alias: 'hotrodlogsparser',
					description: 'Trying to test Logs Pipeline feature',
					enabled: true,
					filter: {
						op: 'AND',
						items: [
							{
								key: {
									key: 'container_name',
									dataType: DataTypes.String,
									type: 'tag',
									isColumn: false,
									isJSON: false,
								},
								id: 'sampleid',
								value: 'hotrod',
								op: '=',
							},
						],
					},
					config: [
						{
							type: 'regex_parser',
							id: 'parsetext(regex)',
							output: 'parseattribsjson',
							on_error: 'send',
							orderId: 1,
							enabled: true,
							name: 'parse text (regex)',
							parse_to: 'attributes',
							regex:
								'.+\\t+(?P<log_level>.+)\\t+(?P<location>.+)\\t+(?P<message>.+)\\t+(?P<attribs_json>.+)',
							parse_from: 'body',
						},
						{
							type: 'json_parser',
							id: 'parseattribsjson',
							output: 'removetempattribs_json',
							orderId: 2,
							enabled: true,
							name: 'parse attribs json',
							parse_to: 'attributes',
							parse_from: 'attributes.attribs_json',
						},
						{
							type: 'remove',
							id: 'removetempattribs_json',
							output: 'c2062723-895e-4614-ba38-29c5d5ee5927',
							orderId: 3,
							enabled: true,
							name: 'remove temp attribs_json',
							field: 'attributes.attribs_json',
						},
						{
							type: 'add',
							id: 'c2062723-895e-4614-ba38-29c5d5ee5927',
							orderId: 4,
							enabled: true,
							name: 'test add ',
							field: 'resource["container.name"]',
							value: 'hotrod',
						},
					],
					createdBy: 'prashant@signoz.io',
					createdAt: '2024-01-02T13:56:02.858300964Z',
				},
				{
					id: '6aea8303-4a62-4c62-ba07-f16ff8f2174a',
					orderId: 2,
					name: 'Logs Parser - Pranay - Customer Service',
					alias: 'LogsParser-Pranay-CustomerService',
					description: 'Trying to test Logs Pipeline feature',
					enabled: true,
					filter: {
						op: 'AND',
						items: [
							{
								key: {
									key: 'service',
									dataType: DataTypes.String,
									type: 'tag',
									isColumn: false,
									isJSON: false,
								},
								id: 'sample-test-1',
								value: 'customer',
								op: '=',
							},
						],
					},
					config: [
						{
							type: 'grok_parser',
							id: 'TestPranay',
							on_error: 'send',
							orderId: 1,
							enabled: true,
							name: 'Test Pranay',
							parse_to: 'attributes',
							pattern:
								'^%{DATE:date}Z INFO customer/database.go:73 Loading customer {"service": "customer", "component": "mysql", "trace_id": "227e184cb1263724", "span_id": "1427a3fcad8b1514", "customer_id": "567"}',
							parse_from: 'body',
						},
					],
					createdBy: 'prashant@signoz.io',
					createdAt: '2024-01-02T13:56:02.863764227Z',
				},
			],
			history: [
				{
					id: '8e196a3c-8737-46d2-8b62-264353b7b9aa',
					version: 24,
					elementType: 'log_pipelines',
					active: false,
					isValid: false,
					disabled: false,
					deployStatus: 'DEPLOYED',
					deployResult: 'Deployment was successful',
					lastHash: 'log_pipelines:24',
					lastConf:
						'[{"id":"1917badc-580c-42fe-a959-cdf895d4f27b","orderId":1,"name":"hotrod logs parser","alias":"hotrodlogsparser","description":"Trying to test Logs Pipeline feature","enabled":true,"filter":{"op":"AND","items":[{"key":{"key":"container_name","dataType":"string","type":"tag","isColumn":false,"isJSON":false},"value":"hotrod","op":"="}]},"config":[{"type":"regex_parser","id":"parsetext(regex)","output":"parseattribsjson","on_error":"send","orderId":1,"enabled":true,"name":"parse text (regex)","parse_to":"attributes","regex":".+\\\\t+(?P\\u003clog_level\\u003e.+)\\\\t+(?P\\u003clocation\\u003e.+)\\\\t+(?P\\u003cmessage\\u003e.+)\\\\t+(?P\\u003cattribs_json\\u003e.+)","parse_from":"body"},{"type":"json_parser","id":"parseattribsjson","output":"removetempattribs_json","orderId":2,"enabled":true,"name":"parse attribs json","parse_to":"attributes","parse_from":"attributes.attribs_json"},{"type":"remove","id":"removetempattribs_json","output":"c2062723-895e-4614-ba38-29c5d5ee5927","orderId":3,"enabled":true,"name":"remove temp attribs_json","field":"attributes.attribs_json"},{"type":"add","id":"c2062723-895e-4614-ba38-29c5d5ee5927","orderId":4,"enabled":true,"name":"test add ","field":"resource[\\"container.name\\"]","value":"hotrod"}],"createdBy":"prashant@signoz.io","createdAt":"2024-01-02T13:56:02.858300964Z"},{"id":"6aea8303-4a62-4c62-ba07-f16ff8f2174a","orderId":2,"name":"Logs Parser - Pranay - Customer Service","alias":"LogsParser-Pranay-CustomerService","description":"Trying to test Logs Pipeline feature","enabled":true,"filter":{"op":"AND","items":[{"key":{"key":"service","dataType":"string","type":"tag","isColumn":false,"isJSON":false},"value":"customer","op":"="}]},"config":[{"type":"grok_parser","id":"TestPranay","on_error":"send","orderId":1,"enabled":true,"name":"Test Pranay","parse_to":"attributes","pattern":"^%{DATE:date}Z INFO customer/database.go:73 Loading customer {\\"service\\": \\"customer\\", \\"component\\": \\"mysql\\", \\"trace_id\\": \\"227e184cb1263724\\", \\"span_id\\": \\"1427a3fcad8b1514\\", \\"customer_id\\": \\"567\\"}","parse_from":"body"}],"createdBy":"prashant@signoz.io","createdAt":"2024-01-02T13:56:02.863764227Z"}]',
					createdBy: '5e9681b1-16c9-4f7a-9a62-48904d73d9c9',
					createdByName: 'Prashant üôé‚Äç‚ôÇÔ∏è',
					createdAt: '2024-01-02T13:56:02Z',
				},
				{
					id: '0b934310-0aea-4806-94b9-7750e3c6935e',
					version: 23,
					elementType: 'log_pipelines',
					active: false,
					isValid: false,
					disabled: false,
					deployStatus: 'DEPLOYED',
					deployResult: 'Deployment was successful',
					lastHash: 'log_pipelines:23',
					lastConf:
						'[{"id":"82051f9e-22f8-4d9d-bca7-35bd32715697","orderId":1,"name":"hotrod logs parser","alias":"hotrodlogsparser","description":"Trying to test Logs Pipeline feature","enabled":true,"filter":{"op":"AND","items":[{"key":{"key":"container_name","dataType":"string","type":"tag","isColumn":false,"isJSON":false},"value":"hotrod","op":"="}]},"config":[{"type":"regex_parser","id":"parsetext(regex)","output":"parseattribsjson","on_error":"send","orderId":1,"enabled":true,"name":"parse text (regex)","parse_to":"attributes","regex":".+\\\\t+(?P\\u003clog_level\\u003e.+)\\\\t+(?P\\u003clocation\\u003e.+)\\\\t+(?P\\u003cmessage\\u003e.+)\\\\t+(?P\\u003cattribs_json\\u003e.+)","parse_from":"body"},{"type":"json_parser","id":"parseattribsjson","output":"removetempattribs_json","orderId":2,"enabled":true,"name":"parse attribs json","parse_to":"attributes","parse_from":"attributes.attribs_json"},{"type":"remove","id":"removetempattribs_json","output":"c2062723-895e-4614-ba38-29c5d5ee5927","orderId":3,"enabled":true,"name":"remove temp attribs_json","field":"attributes.attribs_json"},{"type":"add","id":"c2062723-895e-4614-ba38-29c5d5ee5927","orderId":4,"enabled":true,"name":"test add ","field":"attributes.test","value":"EXPR((attributes.temp?.request_context?.scraper ?? [nil])[0])"}],"createdBy":"prashant@signoz.io","createdAt":"2023-12-29T12:59:20.628696628Z"},{"id":"c9076864-788a-4c5d-bc91-0ab44bbf0b87","orderId":2,"name":"Logs Parser - Pranay - Customer Service","alias":"LogsParser-Pranay-CustomerService","description":"Trying to test Logs Pipeline feature","enabled":true,"filter":{"op":"AND","items":[{"key":{"key":"service","dataType":"string","type":"tag","isColumn":false,"isJSON":false},"value":"customer","op":"="}]},"config":[{"type":"grok_parser","id":"TestPranay","on_error":"send","orderId":1,"enabled":true,"name":"Test Pranay","parse_to":"attributes","pattern":"^%{DATE:date}Z INFO customer/database.go:73 Loading customer {\\"service\\": \\"customer\\", \\"component\\": \\"mysql\\", \\"trace_id\\": \\"227e184cb1263724\\", \\"span_id\\": \\"1427a3fcad8b1514\\", \\"customer_id\\": \\"567\\"}","parse_from":"body"}],"createdBy":"prashant@signoz.io","createdAt":"2023-12-29T12:59:20.635863965Z"}]',
					createdBy: '5e9681b1-16c9-4f7a-9a62-48904d73d9c9',
					createdByName: 'Prashant üôé‚Äç‚ôÇÔ∏è',
					createdAt: '2023-12-29T12:59:20Z',
				},
			],
		};

		const { asFragment } = render(
			<MemoryRouter>
				<QueryClientProvider client={queryClient}>
					<Provider store={store}>
						<I18nextProvider i18n={i18n}>
							<ChangeHistory pipelineData={pipelinedata} />
						</I18nextProvider>
					</Provider>
				</QueryClientProvider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
