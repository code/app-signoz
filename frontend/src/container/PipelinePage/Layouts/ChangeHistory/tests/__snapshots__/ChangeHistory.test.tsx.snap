// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeHistory test should render changeHistory correctly 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 3rem;
}

.c1 {
  padding: 0.625rem;
}

<div
    class="c0"
  >
    <div
      class="ant-table-wrapper css-dev-only-do-not-override-2i2tap"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-2i2tap"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: fixed;"
                >
                  <colgroup />
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Version
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Deployment Stage
                      </th>
                      <th
                        class="ant-table-cell ant-table-cell-ellipsis"
                        scope="col"
                        title="Last Deploy Message"
                      >
                        Last Deploy Message
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Last Deployed Time
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Edited by
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="8e196a3c-8737-46d2-8b62-264353b7b9aa"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        24
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <span
                          aria-label="check-circle"
                          class="anticon anticon-check-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="check-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                            />
                          </svg>
                        </span>
                        <span
                          class="c1"
                        >
                          Deployed
                        </span>
                      </td>
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                        title="Deployment was successful"
                      >
                        Deployment was successful
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <span>
                          January 02, 2024 07:26 PM
                        </span>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Prashant üôé‚Äç‚ôÇÔ∏è
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="0b934310-0aea-4806-94b9-7750e3c6935e"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        23
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <span
                          aria-label="check-circle"
                          class="anticon anticon-check-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="check-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                            />
                          </svg>
                        </span>
                        <span
                          class="c1"
                        >
                          Deployed
                        </span>
                      </td>
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                        title="Deployment was successful"
                      >
                        Deployment was successful
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <span>
                          December 29, 2023 06:29 PM
                        </span>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Prashant üôé‚Äç‚ôÇÔ∏è
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <ul
            class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-2i2tap"
          >
            <li
              aria-disabled="true"
              class="ant-pagination-prev ant-pagination-disabled"
              title="Previous Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="left"
                  class="anticon anticon-left"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
              tabindex="0"
              title="1"
            >
              <a
                rel="nofollow"
              >
                1
              </a>
            </li>
            <li
              aria-disabled="true"
              class="ant-pagination-next ant-pagination-disabled"
              title="Next Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="right"
                  class="anticon anticon-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    />
                  </svg>
                </span>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
